packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -134,6 +134,16 @@ export class DockerImageAsset extends cdk.Construct implements assets.IAsset {
   }
 }
 
+function validateProps(props: DockerImageAssetProps) {
+  for (const [key, value] of Object.entries(props)) {
+    if (Token.isUnresolved(value)) {
+      throw new Error(`Cannot use Token as value of '${key}': this value is used before deployment starts`);
```

Comment:
Would be sweet to have some info about what it's used for before deployment starts...